src/VersionParser.php

```diff
@@ -106,6 +106,13 @@ public function normalize($version, $fullVersion = null)
             $fullVersion = $version;
         }
 
+	    if (preg_match('{^(\^[^,\s]++) ++as ++([^,\s\^]++)$}', $version, $match) ||
+	        preg_match('{^([^,\s]++) ++as ++(\^[^,\s\^]++)$}', $version, $match) ||
+	        preg_match('{^([^,\s]++) ++as ++(\~[^,\s\^]++)$}', $version, $match) ||
+	        preg_match('{^(\~[^,\s]++) ++as ++([^,\s\^]++)$}', $version, $match)) {
```

Comment:
`^` is not the only operator making it a range constraint

Reply:
@stof `^` and `~`. I guess we need to cover both, no?

Reply:
@stof I'm sorry, got it now. Please check the new commit. Thanks!